pkg/tsdb/postgres/postgres.go

```diff
@@ -47,20 +48,53 @@ func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndp
 	timescaledb := datasource.JsonData.Get("timescaledb").MustBool(false)
 
 	endpoint, err := sqleng.NewSqlQueryEndpoint(&config, &queryResultTransformer, newPostgresMacroEngine(timescaledb), logger)
-	if err == nil {
-		logger.Debug("Successfully connected to Postgres")
-	} else {
+	if err != nil {
 		logger.Debug("Failed connecting to Postgres", "err", err)
+		return nil, err
 	}
+
+	logger.Debug("Successfully connected to Postgres")
 	return endpoint, err
 }
 
 func generateConnectionString(datasource *models.DataSource, logger log.Logger) (string, error) {
+	logger.Debug("Trying to generate Postgres connection string")
+
 	sslMode := strings.TrimSpace(strings.ToLower(datasource.JsonData.Get("sslmode").MustString("verify-full")))
 	isSSLDisabled := sslMode == "disable"
 
-	// Always pass SSL mode
-	sslOpts := fmt.Sprintf("sslmode=%s", url.QueryEscape(sslMode))
+	reHost := regexp.MustCompile(`^([^/].*(?::\d+)?)|(/.*)$`)
+	ms := reHost.FindStringSubmatch(datasource.Url)
```

Comment:
Sorry, but this solution feels a bit messy. Having hard time reading and understanding what the code actually does.  Any possibility to use same/similar approach as [mysql](https://github.com/grafana/grafana/blob/5d1508ed8c164cd9d5f054e02290cbcfd491f920/pkg/tsdb/mysql/mysql.go#L34-L45)?

Reply:
@marefr It is a fairly uncomplicated regex to be honest, so I'm not really on board with it being messy. In my opinion the MySQL solution is actually more difficult to understand and verify since it encodes the connection string as a URL, and has to encode every parameter as a result (which led to the bug this PR is fixing in the first place).

I can rewrite the Unix socket detection though, by using strings.HasPrefix as in the MySQL code.